src/game/models/models.test.js

```diff
@@ -1,28 +1,36 @@
-import models from './index'
 import {
+  when,
+  is,
+  uniq,
+  pipe,
+  values,
+  flatten,
+  prop,
   omit,
   map,
 } from 'ramda'
 
-function findAll(needle, haysack) {
+import models from './index'
+
+
+function findAll (needle, haysack) {
   models[haysack].all.map(hay =>
     hay[needle].map(e =>
       models[needle].find(e.id)))
 }
 
-function findAllPlain(needle, haysack) {
+function findAllPlain (needle, haysack) {
   models[haysack].all.map(hay =>
     hay[needle].map(e =>
       models[needle].find(e)))
 }
 
 function equipsOnMonsters () {
-  models.monsters.all.map(monster =>
-    map(
-      prizes => prizes.map(prize =>
-        models.equips.find(prize)),
-      omit(['exp', 'items'], monster.prizes)
-    )
+  const getId = when(is(Object), prop('id'))
+  const getEquipPrizes = pipe(prop('prizes'), omit(['exp', 'items']), values)
+
+  map(prize => models.equips.find(prize),
+    uniq(flatten(models.monsters.all.map(getEquipPrizes)).map(getId)),
```

Comment:
you can also rewrite this with more ramda niceness:

```js
const equipPrizes = pipe(map(getEquipPrizes), map(getId), flatten, uniq)
```

then call it in this line with

```
equipPrizes(models.monsters.all)
```